cli/module.schema.json

```diff
@@ -37,6 +37,21 @@
         },
         "build": {
             "$ref": "#/$defs/build_section"
+        },
+        "overrides": {
+            "type": "object",
+            "description": "Platform overrides that override other fields in meta.json. Keys here take the form 'linux/arm64', see examples. The special 'dev' key applies only in the `module reload` command.",
+            "examples": [
+                {"dev": {"entrypoint": "run.sh"}},
+                {"darwin": {"build.setup": "brew install"}},
+                {"linux/arm64": {"entrypoint": "run-arm.sh"}},
+                {"linux/any": {"entrypoint": "run-linux.sh"}}
+            ],
+            "additionalProperties": {
+                "type": "object",
+                "description": "A map like {\"build.setup\": \"brew install\"}. The keys are dotted paths similar to jsonpath, the values are any replacement value.",
+                "examples": [{"entrypoint": "run.sh"}, {"build.setup": "brew install"}]
+            }
```

Comment:
Sorry that I'm late to reviewing this idea. A few concerns:
- Lets get approval for it (I can send the email) since it does vary from the scope that was approved for adding platform-level commands. 
- I think I prefer `local` over `dev` since that matches the rest of the terminology of running a local module (vs registry module)
- Being able to overwrite _all_ fields feels weird. I think some things shouldn't be overwritten (module_id, description, visibility, etc) and could cause some weird behavior (now and in the future if we depend on meta.json in more places).

Reply:
Now that I am thinking more about the changes we need and the scope that was written, I think what I proposed in the scope isn't enough, but feel like it would be better to have a new section than overwrites.

Either you have the existing `entrypoint` and `build` sections or:

"environments": {
    "local": {"entrypoint":"path.sh"},
    "build:" {"linux/arm64":{"build":"build.sh","setup":"setup.sh","entrypoint":"run.sh"}},
}

A more significant change to the schema but puts us more in the right direction long term. If we use maps of maps and lists of maps everywhere we can easily add additional properties later. The fields we added early on don't have

Reply:
re 'local' -- this applies in remote reloading as well. I'm open to a different name change though

re overwriting module_id -- remember this only applies in reloading mode. a user could just as easily edit meta.json by hand

yup resubmit; I changed where it lives (so that it can modify entrypoint) and the dev key